edb/edgeql/compiler/stmt.py

```diff
@@ -484,6 +490,9 @@ def compile_DeleteQuery(
             context=expr.context,
         )
 
+    # Record this node in the list of potential DML expressions.
+    ctx.env.options.dml_exprs.append(expr)
```

Comment:
It seems to me that a future-proof version of this should be in the form of an inferer-visitor, like the one we have for volatility.  That way we can infer if a chain of function calls is, in fact, a mutation.

Reply:
The benefit of the current approach is that it allows us to mark a specific piece of code as the culprit in the error or warning. Visitors that infer this for the whole subtree will just broadly state whether the body is or isn't DML-containing, which, in case of this being due to another DML-containing function call is not as useful.

Reply:
Nothing prevents the mutation detection visitor to return a list of exprs instead of a boolean answer.

Reply:
In this case I don't really see the benefit of having a separate visitor when this can be done during compile pass. Additionally, if/when we allow DML-containing functions to exist, then we'll need to correctly resolve the (potentially aliased in `WITH` block) name, etc. to correctly infer whether a given expression is DML-containing. This would be just duplication of the same code form compiling function calls.